### DISCUSSION TITLE
Custom module: select all entries from section with Entry::find() extremely slow/not working

### DISCUSSION THREAD

**Discussion by dolphiq-allard** [craftcms/cms#11512] [Q&A]
I am trying to create a custom module where I fill a dropbox with all entries from a specific section (which currently holds 1 entry) using the following code:

```php
use craft\fields\Dropdown;
use craft\elements\Entry;
use craft\elements\db\EntryQuery;

class ThemeSelector extends Dropdown
{
    public static function displayName(): string
    {
        return 'Color theme';
    }

    public function init(): void
    {
        $this->setTheme();
        parent::init();
    }

    protected function setTheme(): void
    {

        $this->options = [
            [
                'label' => 'Choose a color theme',
                'value' => '',
                'disabled' => true,
            ],
        ];

        $themes = Entry::find()
            ->section('colorThemes')
            ->all();

        foreach ($themes as $key => $theme) {
            $this->options[] = [
                'label' => 'Label goes here',
                'value' => 'Value goes here',
            ];
        }

    }


    public function getSettingsHtml(): ?string
    {
        return null;
    }
}
```

However, adding `->all()` to the entry class takes forever and yields a `502 bad gateway` error. If I do the same in a twig template using e.g. `{% for entry in craft.entries.section('colorThemes').all() %}` it works fine and fast.

Any hints on what it is I am missing here. Thanks.

**Comment by Anubarak** [comment]
You are running in an endless loop because you call this function inside your fields `init` function.
This is a well known topic  https://github.com/craftcms/cms/issues/4944

When you execute an ElementQuery while Craft isn't fully bootstrapped it tries to grab all fields in order to append all `where`  and `select` operations for the Query.
So Craft will initialize all fields, and in your init function it will then search for all entries thus create an `ElementQuery`. This one will search for all fields, grabs your field and will make a new `ElementQuery` and so on and so on.

For the sake of performance and best practice you should not fetch all entries unless they are really required... keep in mind otherwise **EVERY** single request will fetch all entries of this section.

**Comment by dolphiq-allard** [comment]
Thanks for pointing me in the right direction. If you don't mind me asking, in your opinion, what would be the correct way to populate my dropdown with the entries from the `colorThemes` section?

**Reply by Anubarak** [reply]
You should do it when you need the entries
For example when you render the options.

You can take a look at the default [Dropdown](https://github.com/craftcms/cms/blob/develop/src/fields/Dropdown.php#L51) field, it passes the available options to the template in the `inputHtml` function because they are not required before either grabbing the field value or before rendering it.

However to be honest I don't fully get why you use a custom Dropdown field and no Entries Field for that purpose to be honest.

**Reply by dolphiq-allard** [reply]
I think it's somewhat more intuitive to present just one selectbox. Furthermore I can combine it with a filter to return a specific piece of code to render the CSS.

Valid point though. Thanks again.
